src/Modal/ModalLayer.jsx

```diff
@@ -52,29 +52,31 @@ function ModalLayer({
     };
   }, [isOpen]);
 
-  if (!isOpen) {
+  if (!isOpen && !isHidden) {
     return null;
   }
 
   const onClickOutside = !isBlocking ? onClose : null;
 
   return (
-    <ModalContextProvider onClose={onClose} isOpen={isOpen} isBlocking={isBlocking}>
+    <ModalContextProvider onClose={onClose} isOpen={isOpen} isHidden={isHidden} isBlocking={isBlocking}>
       <Portal>
         <FocusOn
           allowPinchZoom
           scrollLock
-          enabled={isOpen}
+          enabled={isOpen && !isHidden}
           onEscapeKey={onClose}
           onClickOutside={onClickOutside}
-          className={classNames(
-            'pgn__modal-layer',
-            zIndex ? `zindex-${zIndex}` : '',
-          )}
+          className={classNames({
+            'pgn__modal-layer': isOpen && !isHidden,
+            [`zindex-${zIndex}`]: !!zIndex,
+          })}
         >
           <ModalContentContainer>
-            <ModalBackdrop onClick={onClickOutside} />
-            {children}
+            {!isHidden && <ModalBackdrop onClick={onClickOutside} />}
+            <div className={classNames({ 'd-none': isHidden })}>
```

Comment:
Note: we should avoid using CSS utility classes with `!important` in Paragon components; define a custom `.pgn__` namespaced class name to use instead with `display: none;` to be more themable.